Frends.Community.RabbitMQ/RabbitMQTask.cs

```diff
@@ -86,33 +78,68 @@ public static bool WriteMessage([PropertyTab] WriteInputParams inputParams)
                     basicProperties = _channel.CreateBasicProperties();
                     basicProperties.Persistent = true;
                 }
-                
-                if (inputParams.WriteMessageCount != null &&
-                    _channel.MessageCount(inputParams.QueueName) >= int.Parse(inputParams.WriteMessageCount))
-                {
-                    _channel.TxSelect();
-                    _channel.ConfirmSelect();
-                    _channel.CreateBasicPublishBatch().Publish();
-                    _channel.TxCommit();
-                    if (_channel.MessageCount(inputParams.QueueName) > 0)
-                        _channel.TxRollback();
-                    return true;
-                }
 
-                else
+                
+                if (inputParams.WriteMessageCount == null || 
+                    inputParams.WriteMessageCount != null && int.Parse(inputParams.WriteMessageCount) == 1)
                 {
                     _channel.BasicPublish(exchange:
                         inputParams.ExchangeName,
                         routingKey: inputParams.RoutingKey,
                         basicProperties: basicProperties,
                         body: inputParams.Data);
-                    return false;
+                    return true;
+                }
+                else
+                {
+                    // Add message into a memory based on producer write capability.
+                    if (inputParams.WriteMessageCount != null &&
+                        _channel.MessageCount(inputParams.QueueName) <= int.Parse(inputParams.WriteMessageCount))
+                    {
+                        _channel.CreateBasicPublishBatch().Add(
+                            exchange: inputParams.ExchangeName,
+                            routingKey: inputParams.RoutingKey,
+                            mandatory: true,
+                            properties: basicProperties,
+                            body: inputParams.Data);
+                        return false;
+                    }
+                
+                    // Commit under transaction when all of the messages have been received for the producer.
+                    if (inputParams.WriteMessageCount != null &&
+                        _channel.MessageCount(inputParams.QueueName) == int.Parse(inputParams.WriteMessageCount))
+                    {
+                        try
+                        {
+                            _channel.TxSelect();
+                            _channel.CreateBasicPublishBatch().Publish();
```

Comment:
Doesn't this publish a new empty batch?

Reply:
We have the same publish batch for all of the messages which has been added. In this stage we publish the messages.